pkg/controller/deployment/recreate.go

```diff
@@ -104,8 +106,19 @@ func oldPodsRunning(newRS *extensions.ReplicaSet, oldRSs []*extensions.ReplicaSe
 		if newRS != nil && newRS.UID == rsUID {
 			continue
 		}
-		if len(podList.Items) > 0 {
-			return true
+		for _, pod := range podList.Items {
+			switch pod.Status.Phase {
+			case v1.PodFailed, v1.PodSucceeded:
+				continue
+			case v1.PodUnknown:
+				glog.Warning("Encountered Pod %s/%s in unknown state", pod.Namespace, pod.Name)
```

Comment:
IMO I don't think we need to check unknown pods and print a warning here. Just return `true` in default case should be enough.

Reply:
Agreed. That doesn't conform to the logging standards for controller_manager and could get spammy.

Reply:
you are right - fixed, thx